### PR TITLE
`modcarrier` ot command

### PR DESCRIPTION
**Pull request by canisLupus1313**
Adding `modcarrier` command in openthread diagnostic module.

### CODE REVIEW

File: modules/openthread/platform/radio.c [hunk 1]
```diff
@@ -848,6 +849,33 @@ otError platformRadioTransmitCarrier(otInstance *aInstance, bool aEnable)
 }
 #endif /* CONFIG_IEEE802154_CARRIER_FUNCTIONS */
 
+otError platformRadioTransmitModulatedCarrier(otInstance *aInstance, bool aEnable,
```
Comment (maciejbaczmanski) [inline modules/openthread/platform/radio.c:852]: should this be guarded with some configs? like `CONFIG_IEEE802154_CARRIER_FUNCTIONS` ?

File: modules/openthread/platform/diag.c [hunk 2]
```diff
@@ -276,3 +285,30 @@ otError otPlatDiagGpioGetMode(uint32_t aGpio, otGpioMode *aMode)
...
+
+static otError startModCarrier(otInstance *aInstance, uint8_t aArgsLength, char *aArgs[])
+{
+	ARG_UNUSED(aInstance);
+	ARG_UNUSED(aArgsLength);
+
+	bool enable = true;
+	uint8_t data[OT_RADIO_FRAME_MAX_SIZE + 1];
+
+#if defined(CONFIG_NRF_802154_CARRIER_FUNCTIONS)
```
Comment (maciejbaczmanski) [inline modules/openthread/platform/diag.c:297]:
```suggestion
#if defined(CONFIG_IEEE802154_CARRIER_FUNCTIONS)
```
This part should be platform agnostic, right?

File: modules/openthread/platform/radio.c (hunk 1 above)
Comment (canisLupus1313) [reply modules/openthread/platform/radio.c:851]: fixed. thanks

File: modules/openthread/platform/diag.c (hunk 2 above)
Comment (canisLupus1313) [reply modules/openthread/platform/diag.c]: fixed. thanks

File: modules/openthread/platform/platform-zephyr.h [hunk 3]
```diff
@@ -77,6 +77,12 @@ uint16_t platformRadioChannelGet(otInstance *aInstance);
 otError platformRadioTransmitCarrier(otInstance *aInstance, bool aEnable);
 #endif /* CONFIG_IEEE802154_CARRIER_FUNCTIONS */
 
+/**
+ * Start/stop modulated carrier wave transmission.
+ */
+otError platformRadioTransmitModulatedCarrier(otInstance *aInstance, bool aEnable,
+					      const uint8_t *aData);
+
```
Comment (ankuns) [inline modules/openthread/platform/platform-zephyr.h:85]: bit. This should be wrapped with
#if defined(CONFIG_IEEE802154_CARRIER_FUNCTIONS)
#endif
as the implementation does.

Comment (canisLupus1313) [reply modules/openthread/platform/platform-zephyr.h]: fixed.

**Review by maciejbaczmanski — APPROVED** [review]

**Review by ankuns — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Review by cfriedt — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**